src/lua/init.lua

```diff
@@ -206,8 +206,39 @@ rawset(package, "setsearchroot", setsearchroot)
 local compat = require('compat')
 package.loaded.compat = nil
 
+-- Execute scripts or load modules pointed by TT_PRELOAD
+-- environment variable.
+local function run_preload()
+    local tt_preload = os.getenv('TT_PRELOAD') or ''
+    assert(type(tt_preload) == 'string')
+    if #tt_preload == 0 then
+        return
+    end
+    for _, script in ipairs(tt_preload:split(';')) do
+        -- luacheck: ignore 542 empty if branch
+        if #script == 0 then
+            -- Ignore empty entries to allow duplicated semicolons
+            -- and leading/trailing semicolons.
+            --
+            -- It simplifies construction of the environment
+            -- variable value using concatenation.
+        elseif script:endswith('.lua') then
+            local fn, err = loadfile(script)
+            if fn == nil then
+                error(err)
+            end
+            fn(script)
```

Comment:
Minor: If one wonders what is more convenient way to write the code above, see my implementation:
```lua
local fn = assert(loadfile(script))
fn(script)
```
This can be also folded into one-liner, but it's less convenient and less readable IMHO.

Anyway, feel free to ignore.

Reply:
I don't mind, will do.